### ISSUE TITLE
Ability to modify the query used in listings (entries, etc)

### ISSUE THREAD

**Issue by jasonvarga** [statamic/cms#1895]
Some users have asked to be able to customize what's shown on the entry listing in the CP.
For example, being able to only see entries you've created and hide entries created by other users.

We can override the EntryPolicy's `view` method to prevent them from seeing the entry if they were to visit its URL, but it's still listed.

We should let users customize the query. Maybe being able to throw a closure in a service provider. Something like this:

``` php
EntryIndexQuery::modify(function ($query) {
  $query->where('author', User::current()->id());
});
```

A policy wouldn't work for the listing because we'd need to retrieve all the entries and filter them in PHP, which would screw up pagination etc.

**Comment by jasonvarga** [comment]
An alternative is to use an auto-applying filter. In this example maybe a "My Entries" filter.

The downside would be that you can clear the filter, so you'd end up being able to see all entries again. That may or may not be a problem for you, especially if you also have the policy in place.

**Comment by johncarter-** [comment]
Could provide an example of how to implement the Entry Policy override? I have this in my `AuthServiceProvider`:

```php
    protected $policies = [
        // \Statamic\Contracts\Entries\Entry::class => \Statamic\Policies\EntryPolicy::class,
        \Statamic\Contracts\Entries\Entry::class => App\Policies\PropertyPolicy::class,
    ];
```

And this in the policy
```php
    public function __construct()
    {
        info('yo!');
    }
```

I thought by hitting the collection listing page in the CP I would log a "yo!" but no.

**Comment by Rovaski** [comment]
Would this also be able to be applied to Assets? 
While I can stop users from deleting/moving/editing assets that aren't theirs, I'd like to just prevent them from being shown at all if they aren't that user's (or they don't have the relevant permission).

**Comment by Rovaski** [comment]
How could one filter a query based on whether or not a user "can('view')". 
The current suggestion, comparing for author to current user works fine, but then you'll have to add in extra code to check whether or not the current user is an admin and should be able to see all things or not.

**Comment by johncarter-** [comment]
Did #3053 make this possible?

**Comment by jasonvarga** [comment]
Nope, this is still a valid feature request. 3053 only controls whether entries are editable. They'll still appear in the listings.
